DEPLOYMENT-OPTIONS.md

```diff
@@ -0,0 +1,199 @@
+
+## 🎯 Which Method Should You Choose?
+
+### Use **Helm Chart** if you:
+- Want a production-ready solution
+- Need parameterized configurations
+- Plan to upgrade/manage multiple environments
+- Need dependency management (cert-manager)
+- Want advanced templating capabilities
+- Are comfortable with Helm
+
+### Use **Direct Manifests** if you:
+- Want the simplest deployment
+- Prefer direct kubectl over Helm
+- Have basic configuration needs
+- Are learning Kubernetes
+- Want minimal tooling dependencies
+- Follow GitOps practices with simple deployments
+
+## 🔧 Configuration Examples
+
+### Helm Configuration
+```yaml
+# values.yaml
+squidExporter:
+  enabled: true
+  port: 9301
+  resources:
+    requests:
+      cpu: 10m
+      memory: 16Mi
+    limits:
+      cpu: 100m
+      memory: 64Mi
+
+prometheus:
+  serviceMonitor:
+    enabled: true
+    interval: 30s
+    scrapeTimeout: 10s
+
+namespace:
+  name: proxy
+
+image:
+  repository: localhost/konflux-ci/squid
+  tag: "latest"
+```
+
+### Direct Manifest Customization
+```bash
+# To customize the direct manifest, edit squid-with-monitoring.yaml
+# For example, to change resource limits:
+vim squid-with-monitoring.yaml
+
+# Find the squid-exporter container and modify:
+resources:
+  limits:
+    cpu: 200m      # Change from 100m
+    memory: 128Mi  # Change from 64Mi
+  requests:
+    cpu: 20m       # Change from 10m
+    memory: 32Mi   # Change from 16Mi
+```
+
+## 🚀 Quick Start Examples
+
+### Helm Deployment
+```bash
+# Build dependencies
+cd squid && helm dependency build && cd ..
+
+# Deploy with monitoring enabled
+helm install squid ./squid \
+  --set squidExporter.enabled=true \
+  --set prometheus.serviceMonitor.enabled=false \
+  --set cert-manager.enabled=false
+
+# Test the deployment
+kubectl port-forward -n proxy svc/squid 3128:3128 9301:9301
+```
+
+### Direct Manifest Deployment
+```bash
+# Deploy everything
+kubectl apply -f squid-with-monitoring.yaml
+
+# Wait for pods to be ready
+kubectl wait --for=condition=Ready pod -l app.kubernetes.io/name=squid -n proxy --timeout=120s
+
+# Test the deployment
+kubectl port-forward -n proxy svc/squid 3128:3128 9301:9301
+```
+
+## 🔍 Testing Your Deployment
+
+Regardless of which method you choose, test your deployment with:
+
+```bash
+# Test proxy functionality
+kubectl run test-client --image=curlimages/curl:latest --rm -it -- \
+    sh -c 'curl --proxy http://squid.proxy.svc.cluster.local:3128 http://httpbin.org/ip'
+
+# Test metrics endpoint
+kubectl port-forward -n proxy svc/squid 9301:9301 &
+curl http://localhost:9301/metrics | grep squid_
+```
+
+## 🧹 Cleanup
+
+### Helm Cleanup
+```bash
+helm uninstall squid
+kubectl delete namespace proxy
+```
+
+### Direct Manifest Cleanup
+```bash
+kubectl delete -f squid-with-monitoring.yaml
+```
+
+Both methods deploy the exact same Squid proxy with Prometheus monitoring capabilities. Choose the method that best fits your workflow and requirements! 
```

Comment:
Thats not exactly accurate.
When we deploy squid using the helm chart we get cert-manager as a dependency but not when using the manifests...